### PR TITLE
feat(etherscan): support library verification

### PR DESCRIPTION
**Pull request by tarrencev**
## Motivation

Support etherscan verification of contracts that depend on libraries

## Solution

Serialize libraries into VerifyContract request payload. I'm new to rust so opening this up for feedback early. Appreciate any feedback and happy to make changes. I plan to add tests.

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-etherscan/src/contract.rs [hunk 1]
```diff
@@ -159,6 +173,19 @@ impl ContractMetadata {
...
+fn libs_serialize<S>(l: &Option<Libraries>, serializer: S) -> std::result::Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let libraries = &l.as_ref().unwrap().libs;
+    let mut map = serializer.serialize_map(Some(libraries.len()))?;
+    for (i, (k, v)) in Vec::from_iter(libraries.iter()).iter().enumerate() {
+        map.serialize_entry(&format!("libraryname{}", i), &k.split(":").last())?;
+        map.serialize_entry(&format!("libraryaddress{}", i), v)?;
+    }
```
Comment (mattsse) [inline ethers-etherscan/src/contract.rs:185]: this should only serialize `if let Some(ref libs) =  l = {...}`
could you an example of what this should serialize to?

Comment (tarrencev) [reply ethers-etherscan/src/contract.rs:185]: thanks, appreciate the feedback! still trying to understand control flow in rust. i'll update this to reflect your suggestion. i will also add a basic serialization test as an example. just need to find some time 😅

**Comment by gakonst** [comment]
Bump @tarrencev any interest in pushing this PR over the line?

**Comment by tarrencev** [comment]
> Bump @tarrencev do you still intend in pushing this PR over the line?

hey yeah i would still like to. just havent had much time. if it is blocking something else feel free to take it over. also ok to close for now and i can reopen once i have time to address the issues

**Pull request closed without merging**